api/client/attach.go

```diff
@@ -30,25 +30,29 @@ func (cli *DockerCli) CmdAttach(args ...string) error {
 		return err
 	}
 
-	env := engine.Env{}
-	if err := env.Decode(stream); err != nil {
+	var c struct {
```

Comment:
I couldn't put the struct in types because runconfig depends on opts that depends on api again to validate host, I have a pull request "fixing" this but the support to the old raw struct still remains and I can't put it in types because it  depends on daemon (and daemon has dependency on api/types)  (hope you understand what I wrote lol)